app/models/listing.rb

```diff
@@ -6,4 +6,28 @@ class Listing < ApplicationRecord
   has_many :currencies_listings
   has_many :currencies, through: :currencies_listings
   belongs_to :submitter, class_name: :User, inverse_of: :submissions
+
+  def add_new_currencies(new_currencies)
+    new_currencies.each do |currency|
+      # Currency is live on the listing
+      next if currencies.include? currency
+
+      # Currency was previously on listing but was soft_deleted
+      if currencies_listings.with_deleted
+                            .pluck(:currency_id).include? currency.id
+        # Undelete
+        currencies_listings.with_deleted
+                           .find_by(currency_id: currency.id).recover
+      else
+        currencies << currency
+      end
+    end
+  end
```

Comment:
@mockdeep I excluded from Rubocop. Feel free to refactor. This was complicated enough dealing with acts_as_paranoid and Postgres unique indexes.

Reply:
@dladowitz I wonder if we even need this logic. I think it might make more sense to replace the currencies on the model rather than adding new ones. In that case you just need to say `listing.currencies = currencies`. Also, wondering why we want to use `acts_as_paranoid` for the join records. They're pretty easy to replace, so seems like we're putting a lot of extra work on ourselves to restore them.